test/libdl.jl

```diff
@@ -25,9 +25,10 @@ cd(dirname(@__FILE__)) do
 # @test !isempty(Libdl.find_library(["libccalltest"], [dirname(@__FILE__)]))
 
 # dlopen should be able to handle absolute and relative paths, with and without dlext
+private_libdir = dirname(abspath(Libdl.dlpath("libjulia")))
```

Comment:
I opted to just take the `dlpath` of `"libjulia-debug"` if we're running in debug mode instead, since `libccalltest.so` might be a directory above `sys.so` or not, depending on whether we've run `make install` or not.